kie-server-parent/kie-server-services/kie-server-services-jbpm/src/test/java/org/kie/server/services/jbpm/ConvertUtilsTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.jbpm;
+
+import java.util.Date;
+
+import org.jbpm.kie.services.impl.model.UserTaskInstanceDesc;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.server.api.model.instance.TaskInstance;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConvertUtilsTest {
+
+    @Test
+    public void testConvertToTask() {
+        UserTaskInstanceDesc userTaskWithSla =
+                new UserTaskInstanceDesc(Long.valueOf(1), "Ready", new Date(), "Task1_name", "Task1_desc", 1,
+                                         "Task1_Owner", "Task1_Creator", "deployment", "Process", 1L, new Date(),
+                                         new Date(), 1L, new Date(), ProcessInstance.SLA_PENDING);
+        UserTaskInstanceDesc userTask =
```

Comment:
There's a minimal chance, that the test will be run around midnight, then the various new Date() calls might fall into different days, effectively leading to failing the test.
Could we transform into dates based on string representation? Also to make sure the Date fields are not swapped, I'd use different dates in the string for each field.

Reply:
@jstastny-cz Thanks, but in this case, I don't think that have any issue with the date value regarding the test just checks that the fields have the same date values in both objects

Reply:
Thing is that not all 'new Date()' invocation have to return the same date value (depends on exact time when the constructor is being invoked) .... it is an extreme case, but it's better to prevent it by sticking to fixed values in tests.

Reply:
@jstastny-cz Yes, I think I understand you but for this test I don't care the date value, The test is just checking they have the same value. Anyway, I'm gone to update this test to use a single new Date()

Reply:
@jstastny-cz updated

Reply:
@nmirasch thanks for the change, looks good now.
> The test is just checking they have the same value
Still the values could have been different, cause the Date instances in userTaskWithSla and userTask were instantiated in different moment (possibly slipping over midnight) and the test asserts equality over those. I am only trying to illustrate my concern, I've seen tests 'randomly' failing due to this exact issue, thanks again for the change.

Reply:
@jstastny-cz The test don't compare userTaskWithSla with userTask. The comparison is between userTask and its conversion to TaskInstance ConvertUtils.convertToTask(userTask). But it's ok like this, Thanks for the explanations!

Reply:
:-D I am sorry, I should have read the code more carefully.

Reply:
@jstastny-cz absolutely, no problem. A lot of things to review!